### PR TITLE
Camera controls (task30)

### PR DESCRIPTION
**Pull request by MihaelaGrubii**
Scope: Refactor the camera controller in smce-gd

**Task1:** Create a replacement for the InterpolatedCamera class (as it is being deprecated with Godot 4.0 version)

Workflow: Created a CustomInterpolatedCamera based on the InterpolatedCamera.cpp example

**Task2:** Move the CamCtl responsibilities into the camera class itself 
Workflow: Moved the Cam Controller functions lock_cam, free_cam, set_cam_position. _on_free and related primitives to the Camera 

**Task3:** Change the way the locked on-orbit camera position is calculated so that the camera can operate without adding extra nodes into the tree

Resource: 
https://github.com/godotengine/godot/blob/c8859f0463835266e6e9736d71891f81137673d1/scene/3d/interpolated_camera.cpp#L52-L79

Workflow: Deleted the on-orbit camera position calculations. 

**Modified Classes:**
Camera.gd
SmceHud.gd
Master.tscn
World.gd
ControlPane.gd
CamController.gd
**Deleted:**
ControllableCamera.gd
**Created:**
CustomInterpolatedCamera.gd

### CODE REVIEW

File: project/src/utilities/controllable_camera/CustomInterpolatedCamera.gd [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+func set_speed(p_speed: float) -> void:
+    speed = p_speed;
+
+func get_speed() -> float:
+    return speed;
+
+func _physics_process(delta) -> void:
+        if !enabled:
+                break;
+	    if ! target:
```
Comment (AtiyeEtemadi) [inline project/src/utilities/controllable_camera/CustomInterpolatedCamera.gd]: is space needed between ! and target?

File: project/src/attachments/Camera.gd [hunk 2]
```diff
@@ -37,6 +37,15 @@ var fps = 0
 var vflip: bool = false
 var hflip: bool = false
 
+signal cam_locked
+signal cam_freed
+
+var locked_cam: Spatial = null
```
Comment (AtiyeEtemadi) [inline project/src/attachments/Camera.gd:43]: might not to be lock_cam ?

File: project/src/utilities/controllable_camera/CustomInterpolatedCamera.gd (hunk 1 above)
Comment (MihaelaGrubii) [reply project/src/utilities/controllable_camera/CustomInterpolatedCamera.gd]: nope, I will delete it

File: project/src/attachments/Camera.gd (hunk 2 above)
Comment (MihaelaGrubii) [reply project/src/attachments/Camera.gd:43]: locked_cam is the original variable name used in the CamControls, so I just kept it as is.

**Pull request merged**